src/main/java/org/sonarlint/intellij/promotion/PromotionProvider.kt

```diff
@@ -0,0 +1,157 @@
+/*
+ * SonarLint for IntelliJ IDEA
+ * Copyright (C) 2015-2023 SonarSource
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonarlint.intellij.promotion
+
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.fileEditor.FileEditorManagerListener
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.time.temporal.ChronoUnit
+import org.sonarlint.intellij.common.util.SonarLintUtils
+import org.sonarlint.intellij.config.Settings
+import org.sonarlint.intellij.notifications.SonarLintProjectNotifications
+import org.sonarsource.sonarlint.core.commons.Language
+
+@Service(Service.Level.PROJECT)
+class PromotionProvider(private val project: Project) {
+
+    private val pattern = "hh:mm:ss a, EEE M/d/uuuu"
+    private val dateTimeFormatter = DateTimeFormatter.ofPattern(pattern)
+
+    private val promotedAdvancedLanguages = listOf(Language.JAVA, Language.PYTHON, Language.PHP, Language.JS, Language.TS, Language.CS)
+
+    private val promotedLanguages = listOf(Language.SCALA, Language.PLSQL, Language.SWIFT)
```

Comment:
The names are very close. Maybe `languagesHavingAdvancedRules` and `extraLanguagesInConnectedMode` would be clearer? By the way the second one should not be a static list, PLSQL support depends on the IDE